lib/widgets/content.dart

```diff
@@ -20,6 +20,75 @@ import 'message_list.dart';
 import 'store.dart';
 import 'text.dart';
 
+/// A central place to define styles for Zulip content (rendered Zulip Markdown).
+///
+/// Styles belong here if they differ between light and dark themes,
+/// so that they will animate smoothly (with help from [lerp]) on theme changes.
+/// Styles also belong here if we want to centralize computing them,
+/// for performance. (The message list is particularly performance-sensitive.)
+///
+/// Content elements are assumed to be painted on a theme-appropriate
+/// background. For what this is in the message list, see
+/// widgets/message_list.dart.
+class ContentTheme extends ThemeExtension<ContentTheme> {
+  /// The complete [TextStyle] we use for plain, unstyled paragraphs.
+  ///
+  /// Also the base style that all other text content should inherit from.
+  ///
+  /// This is the complete style for plain paragraphs. Plain-paragraph content
+  /// should not need styles from other sources, such as Material defaults.
+  final TextStyle textStylePlainParagraph;
+
+  ContentTheme(BuildContext context) :
+    textStylePlainParagraph = TextStyle(
+      inherit: false,
+
+      color: const HSLColor.fromAHSL(1, 0, 0, 0.15).toColor(),
+      fontSize: kBaseFontSize,
+      letterSpacing: 0,
+      height: (22 / kBaseFontSize),
+      leadingDistribution: TextLeadingDistribution.even,
+      decoration: TextDecoration.none,
+      fontFamily: kDefaultFontFamily,
+      fontFamilyFallback: defaultFontFamilyFallback,
+    )
+      .merge(weightVariableTextStyle(context))
+      .copyWith(debugLabel: 'ContentTheme.textStylePlainParagraph');
+
+  ContentTheme._({
+    required this.textStylePlainParagraph,
+  });
+
+  @override
+  ContentTheme copyWith({
+    TextStyle? textStylePlainParagraph,
+  }) {
+    return ContentTheme._(
+      textStylePlainParagraph: textStylePlainParagraph ?? this.textStylePlainParagraph,
+    );
+  }
+
+  @override
+  ContentTheme lerp(ContentTheme? other, double t) {
+    if (identical(this, other)) {
+      return this;
+    }
+    return ContentTheme._(
+      textStylePlainParagraph: TextStyle.lerp(textStylePlainParagraph, other?.textStylePlainParagraph, t)!,
+    );
+  }
+
+  /// The [ContentTheme] from the context's active theme.
+  ///
+  /// The [ThemeData] must include [ContentTheme] in [ThemeData.extensions].
+  static ContentTheme of(BuildContext context) {
```

Comment:
nit: let's follow the Flutter conventional order for class members:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#order-other-class-members-in-a-way-that-makes-sense